### DISCUSSION TITLE
Adding folders to Originals folder

### DISCUSSION THREAD

**Discussion by b-onc** [photoprism/photoprism#3754] [Q&A] [answered]
Hello,

After a couple attempts to setup my photoprism instance, I think it's finally good, no broken thumbnails 👍 

Now I want to stop the docker container, shutdown my raspi4, add more folders to the `originals` folder I've mounted, boot up raspberry and start photoprism. What I fear is that I'll have to re-index all photos again, which takes 3 days.

Is there a correct way of doing this so nothing goes wrong for me and I don't lose my thumbnails?

My `originals` folder is on a external HDD which is auto mounted on boot. I've my media files on it and plex server works as expected.

**Comment by graciousgrey** [comment, marked as answer]
In case all folders (originals, import, storage) are mounted properly on persistent volumes you should not loose data when stopping the docker container. 

https://docs.photoprism.app/getting-started/docker-compose/#volumes

You can add additional files to your originals folder at all times you don't need to stop PhotoPrism for this. After you are done adding files simply re-index your files. If you don't select "complete rescan" only newly added or modified files will be indexed.

https://docs.photoprism.app/user-guide/library/originals/

Make sure to not start indexing when PhotoPrism has no access to the originals folder.

**Reply by b-onc** [reply]
Thanks! Does the same apply if I modify the folder structure? So if I move some files that's already been indexed, will they need to be re-indexed?

**Reply by graciousgrey** [reply]
Yes, if you move/rename files PhotoPrism will pick up these changes during the next indexing (again no complete rescan required).
